### ISSUE TITLE
clang-tidy: misc-use-internal-linkage false positive in header without extension

### ISSUE THREAD

**Issue by adriankwiat** [llvm/llvm-project#225529]
In header which has file name without extension, `misc-use-internal-linkage` shows warning on definitions.

`main.cpp`:
```cpp
#include "with.hpp"
#include "without"
int main() {}
```

`with.hpp`:
```cpp
#pragma once
class C1 {};
inline void f1() {}
```

`without`:
```cpp
#pragma once
class C2 {};
inline void f2() {}
```

output of `clang-tidy -checks=misc-use-internal-linkage main.cpp --`:

```
2 warnings generated.
/home/adrian/Projects/test-cpp/without:2:7: warning: class 'C2' can be moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage]
    2 | class C2 {};
      |       ^
/home/adrian/Projects/test-cpp/without:3:13: warning: function 'f2' can be made static or moved into an anonymous namespace to enforce internal linkage [misc-use-internal-linkage]
    3 | inline void f2() {}
      |             ^
      | static 
```

**Comment by EugeneZelenko** [comment]
@adriankwiat: Please use text in code blocks instead of screenshots.
